packages/react-native/src/Authenticator/hooks/useFieldValues/utils.ts

```diff
@@ -39,9 +39,20 @@ export const getSanitizedRadioFields = (
       return false;
     }
 
+    let censoredVal = value;
+    if (name === 'email') {
+      const splitEmail = value.split('@');
+      const censoredName = censorAllButFirstAndLast(splitEmail[0]);
+
+      censoredVal = `${censoredName}@${splitEmail[1]}`;
+    } else if (name === 'phone') {
+      censoredVal = censorPhoneNumber(value);
+    }
```

Comment:
Have we tested the verify contact flow with the values handled this way? Think that it may fail since we use `value` in the submit handler?

Reply:
Moved logic to subcomponent.